service/src/main/kotlin/fi/espoo/evaka/shared/security/AccessControl.kt

```diff
@@ -389,7 +412,7 @@ WHERE employee_id = :userId
             is AuthenticatedUser.Employee -> when (action) {
                 Action.AssistanceAction.READ_PRE_PRESCHOOL_ASSISTANCE_ACTION ->
                     // If child is or has been in preschool, do not show pre preschool daycare assistance actions for non admin
-                    if (user.isAdmin) true
+                    if (user.isAdmin || user.roles.contains(UserRole.SPECIAL_EDUCATION_TEACHER)) this.assistanceAction.hasPermission(user, action, id)
```

Comment:
I think this should be:
```
if (user.isAdmin || (user.roles.contains(...) && this.assistanceAction.hasPermission(user, action, id)) return true
```
Otherwise admin doesn't get the permission.

Reply:
Well that hasPermission function does allow everything for admin (there is even a test for that), but I agree it is better to be explicit than hide it underneath

Reply:
Ok, okay. Didn't see that.